src/goto-programs/goto_trace.cpp

```diff
@@ -35,6 +36,12 @@ static optionalt<symbol_exprt> get_object_rec(const exprt &src)
     return get_object_rec(to_index_expr(src).array());
   else if(src.id()==ID_typecast)
     return get_object_rec(to_typecast_expr(src).op());
+  else if(
+    src.id() == ID_byte_extract_little_endian ||
+    src.id() == ID_byte_extract_big_endian)
+  {
+    return get_object_rec(to_byte_extract_expr(src).op());
```

Comment:
Remind me, do we actually consider the offset? What you propose is consistent with what's done for index and member expressions, but I'm still wondering whether the overall result is meaningful.

Reply:
The meaningfulness is questionable, but that probably goes beyond this fix.
It might be good to actually specify what these fields are supposed to mean precisely.

Reply:
Here's an example to validate:
````c
#include <assert.h>

int main()
{
  int x;
  char *p = (char *)&x + 2;
  *p = 1;
  assert(x == 0);
}
````
The trace currently includes `byte_extract_little_endian(x, 2l, char)=1 (00000001)` - which might not be downstream-consumable, but at least is accurate. If after this PR the result is `x=1` then there's a problem I'm afraid.

Reply:
goto_trace_steps offer two bits of information:
1) The thing returned by `get_lhs_object()` -- that is the object as defined in the C standard, and `x` is the right answer.
2) The `full_lhs`, which is documented to be "after dereferencing". It is supposed to be exactly what gets assigned. The byte_extract expression above is the best answer for that, in my view.
The trace given to the user should show the latter. We may wish to print byte_extract expressions there as a pointer typecast plus dereferencing.

Reply:
Thank you for clarifying! +1 for printing `byte_extract` as some C expression, which will be important for SV-COMP as well. But I accept that this might be outside the scope of this PR.